src/App.tsx

```diff
@@ -1,18 +1,66 @@
-import { BrowserRouter as Router } from 'react-router-dom';
-import { Provider } from 'react-redux';
+import {
+  RouterProvider,
+  createBrowserRouter,
+  redirect,
+} from 'react-router-dom';
+import { useEffect } from 'react';
 
-import routes from './pages';
-import AppRoutes from './AppRoutes';
-import store from './redux/store';
+import privateRoutes from './pages';
+import { useAppSelector } from './modules/_partials/hooks/useRedux';
+import isAuth from './helpers/isAuth';
+import PageNotFound from './modules/activities/PageNotFound';
+import LoginActivity from './modules/activities/LoginActivity';
+import AppLayout from './modules/_partials/layouts/AppLayout';
+
+const user = isAuth();
+
+const router = createBrowserRouter([
+  {
+    path: '/',
+    errorElement: <PageNotFound />,
+    element: <AppLayout />,
+    loader() {
+      if (!user) {
+        throw redirect('/login');
+      }
+      return <div>Loading</div>;
+    },
+    children: privateRoutes.map(route => ({
+      path: route.path,
+      element: <route.component />,
+    })),
+  },
+  {
+    path: '/login',
+    element: <LoginActivity />,
+    loader() {
+      if (user) {
+        throw redirect('/');
+      }
+      return <div>Loading...</div>;
+    },
+  },
+]);
 
 const App = () => {
-  return (
-    <Provider store={store}>
-      <Router>
-        <AppRoutes routes={routes} />
-      </Router>
-    </Provider>
-  );
+  const { theme } = useAppSelector(state => state.theme);
+
+  useEffect(() => {
```

Comment:
Can we use useLayoutEffect instead of useEffect when toggling dark mode? The reason is that useLayoutEffect runs synchronously after all DOM mutations but before the browser has a chance to paint. This means that it can be used to manipulate the DOM without causing flickering or other visual artifacts. On the other hand, useEffect runs asynchronously after the browser has painted, which can cause flickering or other visual artifacts when manipulating the DOM. Even though it just makes a slight difference, it makes sense as the theme deals with the layout painting.